js/src/pages/settings/enhanced-conversions/useEnableEnhancedConversions.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '~/data';
+import useAppSelectDispatch from '~/hooks/useAppSelectDispatch';
+
+const selectorName = 'getEnableEnhancedConversions';
+
+/**
+ * Custom hook to manage the state of enhanced conversions.
+ * @return {Array} An array containing the current state and a function to toggle it.
+ */
+export const useEnableEnhancedConversions = () => {
+	const { data, isResolving } = useAppSelectDispatch( selectorName );
```

Comment:
@ankitrox Let's avoid using `isResolving` and instead use `hasFinishedResolution`, because `isResolving` will initially be `false` and then `true` when it starts resolving. This will lead to some conditions being triggered when the value of `isResolving` changes. `hasFinishedResolution` will be always `false`.